DigitalLearningSolutions.Data/DataServices/GroupsDataService.cs

```diff
@@ -371,36 +408,19 @@ DateTime timeOfRemoval
         public Group? GetGroupAtCentreById(int groupId, int centreId)
         {
             return connection.Query<Group>(
-                @$"SELECT
-                        GroupID,
-                        GroupLabel,
-                        GroupDescription,
-                        (SELECT COUNT(*) FROM GroupDelegates AS gd WHERE gd.GroupID = g.GroupID) AS DelegateCount,
-                        ({CourseCountSql}) AS CoursesCount,
-                        g.CreatedByAdminUserID As AddedByAdminId,
-                        au.Forename AS AddedByFirstName,
-                        au.Surname AS AddedByLastName,
-                        LinkedToField,
-                        CASE
-                            WHEN LinkedToField = 0 THEN 'None'
-                            WHEN LinkedToField = 1 THEN (SELECT cp.CustomPrompt FROM CustomPrompts AS cp WHERE cp.CustomPromptID = c.CustomField1PromptID)
-                            WHEN LinkedToField = 2 THEN (SELECT cp.CustomPrompt FROM CustomPrompts AS cp WHERE cp.CustomPromptID = c.CustomField2PromptID)
-                            WHEN LinkedToField = 3 THEN (SELECT cp.CustomPrompt FROM CustomPrompts AS cp WHERE cp.CustomPromptID = c.CustomField3PromptID)
-                            WHEN LinkedToField = 4 THEN 'Job group'
-                            WHEN LinkedToField = 5 THEN (SELECT cp.CustomPrompt FROM CustomPrompts AS cp WHERE cp.CustomPromptID = c.CustomField4PromptID)
-                            WHEN LinkedToField = 6 THEN (SELECT cp.CustomPrompt FROM CustomPrompts AS cp WHERE cp.CustomPromptID = c.CustomField5PromptID)
-                            WHEN LinkedToField = 7 THEN (SELECT cp.CustomPrompt FROM CustomPrompts AS cp WHERE cp.CustomPromptID = c.CustomField6PromptID)
-                        END AS LinkedToFieldName,
-                        AddNewRegistrants,
-                        SyncFieldChanges
-                    FROM Groups AS g
-                    JOIN AdminUsers AS au ON au.AdminID = g.CreatedByAdminUserID
-                    JOIN Centres AS c ON c.CentreID = g.CentreID
-                    WHERE RemovedDate IS NULL AND g.CentreID = @centreId AND GroupID = @groupId",
+                @$"{GroupsSql} AND g.CentreID = @centreId AND GroupID = @groupId",
                 new { groupId, centreId }
             ).SingleOrDefault();
         }
 
+        public Group? GetGroupAtCentreByName(string groupName, int centreId)
+        {
+            return connection.Query<Group>(
+                @$"{GroupsSql} AND g.CentreID = @centreId AND GroupLabel = @groupName",
+                new { groupName, centreId }
+            ).SingleOrDefault();
+        }

```

Comment:
I'm not sure about the inclusion of this method, and the logical implication of how it's being used. I think it's quite likely we're overusing calls to the database, when we could just get all the groups at the start of our new service method, and query the list in the C#.

Reply:
Good idea, replaced with one call to GetGroupsForCentre

Reply:
Oops, must have got added back in when I was resolving merge conflicts. It's gone for good now